istio-control/istio-autoinject/templates/configmap.yaml

```diff
@@ -73,19 +73,30 @@ data:
           address: {{ .Values.global.tracer.datadog.address }}
       {{- end }}
 
+    {{- $defPilotHostname := printf "istio-pilot%s.%s" .Values.version .Values.global.configNamespace }}
```

Comment:
one case:  if `.Values.version` is not empty, and `remotePilotCreateSvcEndpoint` is set to true, the pilotAddress would be inconsistent with pilot svc address defined at: https://github.com/istio/installer/blob/master/base/templates/services.yaml#L2-L23
Because the svc created in remote cluster is without `.Values.version`